src/ts/components/listbox.ts

```diff
@@ -0,0 +1,115 @@
+import {Button, ButtonConfig} from './button';
+import {ListSelector, ListSelectorConfig} from './listselector';
+import {DOM} from '../dom';
+
+/**
+ * A element to select a single item out of a list of available items.
+ *
+ * DOM example:
+ * <code>
+ *   <div class='ui-listbox>
+ *     <div class='ui-listboxitem-wrapper>
+ *       <div class='ui-listboxitem'>
+ *         <button class='ui-listboxitem-button'>label</button>
+ *       </div>
+ *     </div>
+ *     ...
+ *   </div
+ * </code>
+ */
+export class ListBox extends ListSelector<ListSelectorConfig> {
+  private listBoxElement: DOM;
+
+  private static readonly SELECTED_LIST_BOX_ITEM_CLASS = 'ui-listboxitem-button-selected';
+
+  constructor(config: ListSelectorConfig = {}) {
+    super(config);
+
+    this.config = this.mergeConfig(config, {
+      cssClass: 'ui-listbox',
+    } as ListSelectorConfig, this.config);
+  }
+
+  protected toDomElement(): DOM {
+    let listBoxElement = new DOM('div', {
+      'id': this.config.id,
+      'class': this.getCssClasses(),
+    });
+
+    this.listBoxElement = listBoxElement;
+    this.updateDomItems();
+
+    return listBoxElement;
+  }
+
+  protected updateDomItems(selectedValue: string = null) {
+    // Delete all children
+    this.listBoxElement.empty();
+
+    // Add updated children
+    for (let item of this.items) {
+      let itemButton = new ListBoxItemButton({
+        key: item.key,
+        text: item.label,
+      });
+
+      itemButton.onClick.subscribe((sender) => {
+        this.handleSelectionChange(<ListBoxItemButton>sender);
+      });
+
+      const selectedItemClass = this.prefixCss(ListBox.SELECTED_LIST_BOX_ITEM_CLASS);
+      const itemElement = itemButton.getDomElement();
```

Comment:
This is actually a hack. To do it right, the `ListSelector` must be converted to extend `Container` and the `SelectBox` needs to be adjusted to that. Currently you are managing `Button` instances outside of the component tree. This might have side effects. But lets let that be for the moment.